src/components/input/SelectNext.tsx

```diff
@@ -60,7 +60,35 @@ function SelectOption<T>({
   }
 
   const { selectValue, value: currentValue } = selectContext;
-  const selected = !disabled && currentValue === value;
+  const valueIsArray = Array.isArray(currentValue);
+
+  const selected =
+    !disabled &&
+    ((valueIsArray && currentValue.includes(value)) || currentValue === value);
+
+  const selectOrToggle = useCallback(() => {
+    // In single-select, just set current value
+    if (!valueIsArray) {
+      selectValue(value);
+      return;
+    }
+
+    // In multi-select, clear selection for nullish values
+    if (!value) {
```

Comment:
As the component now enforces that `value` is an array for a multi-select, this shouldn't be possible any more.

Reply:
`value` is the value passed for the particular option, not the `selectedValue` of the select.

This is used so that we can have something like `<SelectNext.Option value={undefined}>Reset selection</SelectNext.Option>`.

Reply:
[I mentioned](https://github.com/hypothesis/frontend-shared/pull/1601#discussion_r1662624153) this decision is probably too opinionated and confusing, and I think this is proving it.

I'll try to think on a different approach to achieve this, or perhaps delay this to a follow-up PR.

In my first comment I was proposing an alternative approach in the lines of `<SelectNext.Option onSelect={() => setValue([])}>Reset selection</SelectNext.Option>`, where `onSelect` is invoked only if provided, in addition, or perhaps instead, of the regular internal logic implemented in `SelectNext.Option`.

Reply:
> value is the value passed for the particular option, not the selectedValue of the select.

🤦‍♂️ . I follow. OK, the current API design actually seems reasonable to me in that case. Just make sure it is documented.